pkg/kubernetes/components.go

```diff
@@ -22,20 +27,51 @@ type ComponentsOutput struct {
 	Age     string `csv:"AGE"`
 }
 
-// List outputs all Dapr components.
-func Components() ([]ComponentsOutput, error) {
-	client, err := DaprClient()
+// PrintComponents prints all Dapr components.
+func PrintComponents(name, outputFormat string) error {
+	return writeComponents(os.Stdout, func() (*v1alpha1.ComponentList, error) {
+		client, err := DaprClient()
+		if err != nil {
+			return nil, err
+		}
+
+		return client.ComponentsV1alpha1().Components(meta_v1.NamespaceAll).List(meta_v1.ListOptions{})
+	}, name, outputFormat)
+}
+
+func writeComponents(writer io.Writer, getConfigFunc func() (*v1alpha1.ComponentList, error), name, outputFormat string) error {
+	confs, err := getConfigFunc()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	comps, err := client.ComponentsV1alpha1().Components(meta_v1.NamespaceAll).List(meta_v1.ListOptions{})
-	if err != nil {
-		return nil, err
+	filtered := []v1alpha1.Component{}
+	filteredSpecs := []configurationDetailedOutput{}
+	for _, c := range confs.Items {
+		confName := c.GetName()
+		if confName == "daprsystem" {
+			continue
+		}
+
+		if name == "" || confName == name {
```

Comment:
case insensitive?

Reply:
Yeah, I'll apply that across the others too.